core/src/test/java/org/apache/calcite/sql/test/SqlOperatorBaseTest.java

```diff
@@ -8093,6 +8094,23 @@ void checkArrayConcatAggFuncFails(SqlTester t) {
     tester.checkColumnType(
         "select cast(null as any)['x'] from (values(1))",
         "ANY");
+
+    // Row item
+    final String intStructQuery = "select \"T\".\"X\"[1] "
+        + "from (VALUES (ROW(ROW(3, 7), ROW(4, 8)))) as T(x, y)";
+    tester.check(intStructQuery, SqlTests.INTEGER_TYPE_CHECKER, 3, 0);
```

Comment:
Does SQL standard allows access row fields with indices ?

Reply:
I have replied at the ticket comment so that it's more visible.